hexa/core/analytics.py

```diff
@@ -0,0 +1,87 @@
+from datetime import datetime
+
+from django.conf import settings
+from django.http import HttpRequest
+from mixpanel import BufferedConsumer, Mixpanel
+from sentry_sdk import capture_exception
+from ua_parser import user_agent_parser
+
+mixpanel_consumer = BufferedConsumer()
+
+
+def track(
+    user,
+    event: str,
+    properties: dict = {},
+    request: HttpRequest = None,
```

Comment:
Your typing is not correct. You typed the request to be a HttpRequest to which you set the default value to None and not a HttpRequest instance 🤷‍♂️